File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ecs/components/SaveGameComponent.java
package de.limbusdev.guardianmonsters.fwmengine.world.ecs.components;

import com.badlogic.ashley.core.Component;

import de.limbusdev.guardianmonsters.model.gamestate.GameState;


/**
 * Created by georg on 03.12.15.
 */
public class SaveGameComponent implements Component {
    /* ............................................................................ ATTRIBUTES .. */
    public GameState gameState;
    /* ........................................................................... CONSTRUCTOR .. */

    public SaveGameComponent(GameState gameState) {
        this.gameState = gameState;
    }
    /* ............................................................................... METHODS .. */
    
    /* ..................................................................... GETTERS & SETTERS .. */
}

File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/ecs/components/SaveGameComponent.kt
package de.limbusdev.guardianmonsters.fwmengine.world.ecs.components

import de.limbusdev.guardianmonsters.fwmengine.world.ecs.LimbusBehaviour

import de.limbusdev.guardianmonsters.model.gamestate.GameState


/**
 * SaveGameComponent
 *
 * @author Georg Eckert 2015-12-03
 */
object SaveGameComponent : LimbusBehaviour()
{
    var gameState: GameState? = null
    private val checkPoints = mutableSetOf<Int>()

    fun enableCheckPoint(checkPointID: Int)
    {
        checkPoints.add(checkPointID)
    }

    /** Returns true, if the player has already gathered the check point in question. */
    fun isCheckPointEnabled(checkPointID: Int) : Boolean
    {
        return checkPoints.contains(checkPointID)
    }
}